ttnn/cpp/ttnn/operations/pool/rotate/device/rotate_nearest_program_factory.cpp

```diff
@@ -0,0 +1,277 @@
+    const uint32_t l1_for_cb = available_l1 / NEAREST_BUFFERING_FACTOR;
+    const uint32_t max_cb_pages_from_l1 = l1_for_cb / aligned_input_stick_nbytes;
+
+    // Determine actual number of CB pages: min of (max work per core, L1 capacity)
+    const uint32_t max_sticks_per_core = std::max(num_sticks_per_core_group_1, num_sticks_per_core_group_2);
+    uint32_t num_cb_pages = std::min(max_sticks_per_core, max_cb_pages_from_l1);
+    num_cb_pages = num_cb_pages * NEAREST_BUFFERING_FACTOR;
+
+    // CB_0: Output CB for communication between reader and writer
+    const uint32_t output_cb_page_size = aligned_input_stick_nbytes;
+    const auto [output_cb_index, output_cb_handle] = tt::tt_metal::create_cb(
+        tt::CBIndex::c_0, program, all_cores, output_cb_page_size, num_cb_pages, output_cb_data_format);
+
+    // CB_1: Fill CB - single page to hold pre-filled stick for L1-to-L1 copy
+    const auto [fill_cb_index, fill_cb_handle] =
+        tt::tt_metal::create_cb(tt::CBIndex::c_1, program, all_cores, output_cb_page_size, 1, output_cb_data_format);
+
+    // Check if fill value is zero (can use MEM_ZEROS_BASE)
+    const bool fill_is_zero = (fill_value_bf16 == 0);
+
+    // Reader compile-time arguments (RISCV_0)
+    std::vector<uint32_t> reader_compile_time_args = {
+        output_cb_index,                      // ct_arg[0]: output_cb_index
+        aligned_input_stick_nbytes,           // ct_arg[1]: aligned_input_stick_nbytes (for DRAM reads)
+        input_batch,                          // ct_arg[2]: input_batch
+        input_height,                         // ct_arg[3]: input_height
+        input_width,                          // ct_arg[4]: input_width
+        input_channels,                       // ct_arg[5]: input_channels
+        num_cb_pages,                         // ct_arg[6]: num_cb_pages
+        fill_cb_index,                        // ct_arg[7]: fill_cb_index
+        input_stick_nbytes,                   // ct_arg[8]: input_stick_nbytes (unaligned, for fill)
+        static_cast<uint32_t>(fill_is_zero),  // ct_arg[9]: fill_is_zero
+    };
+
+    // Append tensor accessor args for input tensor (starts at ct_arg[10])
+    tt::tt_metal::TensorAccessorArgs(*input_tensor.buffer()).append_to(reader_compile_time_args);
+
+    // Writer compile-time arguments (RISCV_1)
+    std::vector<uint32_t> writer_compile_time_args = {
+        output_cb_index,              // ct_arg[0]: output_cb_index
+        aligned_output_stick_nbytes,  // ct_arg[1]: aligned_output_stick_nbytes
+        num_cb_pages,                 // ct_arg[2]: num_cb_pages
+    };
+
+    // Append tensor accessor args for output tensor (starts at ct_arg[3])
+    tt::tt_metal::TensorAccessorArgs(*output_tensor.buffer()).append_to(writer_compile_time_args);
+
+    // Create reader kernel (RISCV_0)
+    tt::tt_metal::KernelHandle reader_kernel_id = tt::tt_metal::CreateKernel(
```

Comment:
I think it would be better to juse ReaderDataMovementConfig and WriterDataMovementConfig